src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -139,9 +139,34 @@ private void LogSuccessfulTelemetryToClient(AuthenticationResult authenticationR
                 telemetryEventDetails.SetProperty(TelemetryConstants.DurationInCache, authenticationResult.AuthenticationResultMetadata.DurationInCacheInMs);
                 telemetryEventDetails.SetProperty(TelemetryConstants.DurationInHttp, authenticationResult.AuthenticationResultMetadata.DurationInHttpInMs);
                 telemetryEventDetails.SetProperty(TelemetryConstants.Succeeded, true);
-                telemetryEventDetails.SetProperty(TelemetryConstants.PopToken, authenticationResult.TokenType.Equals(Constants.PoPTokenType));
+                telemetryEventDetails.SetProperty(TelemetryConstants.TokenType, (int)AuthenticationRequestParameters.RequestContext.ApiEvent.TokenType);
                 telemetryEventDetails.SetProperty(TelemetryConstants.RemainingLifetime, (authenticationResult.ExpiresOn - DateTime.Now).TotalMilliseconds);
                 telemetryEventDetails.SetProperty(TelemetryConstants.ActivityId, authenticationResult.CorrelationId);
+                telemetryEventDetails.SetProperty(TelemetryConstants.RefreshOn, 
+                    authenticationResult.AuthenticationResultMetadata.RefreshOn.HasValue ?
+                    DateTimeHelpers.DateTimeToUnixTimestampMilliseconds(authenticationResult.AuthenticationResultMetadata.RefreshOn.Value)
+                    : 0);
+                telemetryEventDetails.SetProperty(TelemetryConstants.AssertionType, (int)AuthenticationRequestParameters.RequestContext.ApiEvent.AssertionType);
+                telemetryEventDetails.SetProperty(TelemetryConstants.Endpoint, AuthenticationRequestParameters.Authority.AuthorityInfo.CanonicalAuthority.ToString());
+                telemetryEventDetails.SetProperty(TelemetryConstants.CacheUsed, (int)telemetryDatapoints.CacheTypeUsed);
+                ParseScopesForTelemetry(telemetryEventDetails);
+            }
+        }
+
+        private void ParseScopesForTelemetry(MsalTelemetryEventDetails telemetryEventDetails)
+        {
+            if (AuthenticationRequestParameters.Scope.Count > 0)
```

Comment:
The spec mentions: 
```
resource - the uri
scopes - scopes without the resource uri
```

This is to avoid putting string that are too long into the telemetry storage.

Reply:
We should have a few validation tests for this: 

```
Input: https:\\graph.microsoft.com\.default
Resource: https:\\graph.microsoft.com
Scopes: .default

Input: https:\\graph.microsoft.com\User.Read https:\\graph.microsoft.com\Mail.Read 
Resource: https:\\graph.microsoft.com
Scopes: User.Read Mail.Read

Input: User.Read Mail.Read
Resource: graph
Scopes: User.Read Mail.Read

Input: https:\\sharepoint.com\scope
Resource: https:\\sharepoint.com\
Scopes: scope

Reply:
Thanks, yes, examples help. My first thought top of mind is the concern for all the string parsing. The other scope parsing logic that we have already also shows up on flamegraphs as high CPU usage. Now since this will be done only when telemetry is enabled, maybe it's not too bad; but again if we want telemetry to be used widely, it'll be an issue. This is also done for every request.

This case means I'll have to parse every scope?
```text
Input: https:\\graph.microsoft.com\User.Read https:\\graph.microsoft.com\Mail.Read
Resource: https:\\graph.microsoft.com
Scopes: User.Read Mail.Read
```

Reply:
> We should have a few validation tests for this. Let me know if any of these don't make sense.
> 
> ```
> Input: https:\\graph.microsoft.com\.default
> Resource: https:\\graph.microsoft.com
> Scopes: .default
> 
> Input: https:\\graph.microsoft.com\User.Read https:\\graph.microsoft.com\Mail.Read 
> Resource: https:\\graph.microsoft.com
> Scopes: User.Read Mail.Read
> 
> Input: User.Read Mail.Read
> Resource: 
> Scopes: User.Read Mail.Read
> 
> Input: https:\\sharepoint.com\scope
> Resource: https:\\sharepoint.com\
> Scopes: scope
> 
> Input: null
> Resource: null
> Scopes: null
> 
> Input: any combo of the 3 magic scopes (offline_access, openid, profile)
> Resource: null
> Scopes: null
> ```
> 
> I'm hoping that we don't have to do extra parsing for the last 2 use cases because MSAL alraedy has scope merging logic.
> 
> The use cases above should be covered by tests.

Ah, ok ,this makes sense. ill go ahead and update the logic to reflect this

Reply:
> 



> Thanks, yes, examples help. My first thought top of mind is the concern for all the string parsing. The other scope parsing logic that we have already also shows up on flamegraphs as high CPU usage. Now since this will be done only when telemetry is enabled, maybe it's not too bad; but again if we want telemetry to be used widely, it'll be an issue. This is also done for every request.
> 
> This case means I'll have to parse every scope?
> 
> ```
> Input: https:\\graph.microsoft.com\User.Read https:\\graph.microsoft.com\Mail.Read
> Resource: https:\\graph.microsoft.com
> Scopes: User.Read Mail.Read
> ```

I guess it would depend on whether or not the host portion of the scopes are the same for all scopes. If so, only one scope needs to be parsed and the extracted host can simply be removed from the other scopes with something like `string.Split(host)`. 

However, if we have something like the following then we may have to parse all of them.

 ```
 Input: https:\\hostA\User.Read https:\\HostB\Mail.Read
 Resource: https:\\hostA https:\\hostB
 Scopes: User.Read Mail.Read
 ```
 
 @bgavrilMS Will the host portion of the scopes all be the same?

Reply:
@pmaytak discussed this with Bogdan, We only need to parse one of the Uri and then make the assumption that other scopes will contain the same host

Reply:
@bgavrilMS cant do this test since null cant be used as a scope with confidential client
```
Input: null
Resource: null
Scopes: null
```